saleor/graphql/checkout/mutations.py

```diff
@@ -384,9 +421,22 @@ class Meta:
 
     @classmethod
     def perform_mutation(cls, _root, info, checkout_id, shipping_address):
-        checkout = cls.get_node_or_error(
-            info, checkout_id, only_type=Checkout, field="checkout_id"
+        pk = from_global_id_strict_type(
+            info, checkout_id, Checkout, field="checkout_id"
         )
+
+        try:
+            checkout = models.Checkout.objects.prefetch_related(
+                "lines__variant__product__product_type"
+            ).get(pk=pk)
+        except ObjectDoesNotExist:
+            raise ValidationError(
+                {"checkout_id": f"Couldn't resolve to a node: {checkout_id}"}
+            )
+
+        if not checkout.is_shipping_required():
+            raise ValidationError({"shipping_address": ERROR_DOES_NOT_SHIP})
```

Comment:
`ERROR_DOES_NOT_SHIP = "This checkout doesn't have shipping"` -> Wouldn't the `"This checkout doesn't need shipping"` be better?